src/sensor/protocoldetector.cpp

```diff
@@ -130,6 +130,12 @@ bool ProtocolDetector::checkSerial(LinkConfiguration& linkConf)
     }
     qCDebug(PING_PROTOCOL_PROTOCOLDETECTOR) << "Port is open";
     port.setBaudRate(baudrate);
+    port.sendBreak();
```

Comment:
`sendBreak` is deprecated, please do not use it.

Reply:
I don't see anything about deprecation in 5.13: https://doc.qt.io/qt-5/qserialport.html#sendBreak

Reply:
it was deprecated, then undeprecated. but patrick is right - this is blocking, and thus should be avoided. https://forum.qt.io/topic/90664/qserialport-sendbreak-is-deprecated-but-no-alternative-is-proposed/5

Reply:
This class runs in it's own thread.

- I think that means it should be ok if it is blocking?
- If we use setbreakenabled and QThread::msleep that is still blocking, correct?
- What is the non-blocking alternative?

Reply:
@tcanabrava 	Where did you get that is not deprecated anymore ? Here I have warning messages for both 5.12 and 5.13

Reply:
Guys, any suggestion here to get it merged? This fixes bugs preventing proper operation of the device.

Reply:
This is results in deprecated warnings here, so I'm a bit apprehensive about it.

Reply:
I'll try setbreakenabled with msleep.

Reply:
> This results in deprecated warnings here, so I'm a bit apprehensive about it.

can you share the build log and point the line #?

Reply:
```
../src/link/seriallink.cpp: In member function ‘virtual bool SerialLink::setConfiguration(const LinkConfiguration&)’:
../src/link/seriallink.cpp:50:21: warning: ‘bool QSerialPort::sendBreak(int)’ is deprecated [-Wdeprecated-declarations]
   50 |     _port.sendBreak();
      |                     ^
In file included from /usr/include/qt/QtSerialPort/QSerialPort:1,
                 from ../src/link/seriallink.h:3,
                 from ../src/link/seriallink.cpp:7:
/usr/include/qt/QtSerialPort/qserialport.h:269:24: note: declared here
  269 |     QT_DEPRECATED bool sendBreak(int duration = 0);

```

Reply:
using setbreakenabled now